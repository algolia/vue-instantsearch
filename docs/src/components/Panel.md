---
title: Panel
mainTitle: Components
layout: main.pug
category: Components
withHeadings: true
navWeight: 6
editable: true
githubSource: docs/src/components/Panel.md
---

The Panel widget wraps other widgets in a consistent panel design. It also reacts, indicates and set CSS classes when widgets are no more relevant for refining. E.g. when a RefinementList becomes empty because of the current search results.

<a class="btn btn-static-theme" href="stories/?selectedKind=Panel">ðŸ•¹ try out live</a>

## Usage

```html
<ais-panel>
  <template slot="header">Brand</template>
  <ais-refinement-list attribute="brand" />
</ais-panel>
```

## Props

Name | Type | | Default | Description | Required
---|---|---|---|---
classNames | Object | | Override class names | no

## Slots

name | scope | Description
---|---|---
`header` | - | Adds a header to the widget
`default`Â | - | Adds a body to the widget
`footer` | - | Adds a footer to the widget


## CSS classes

Here's a list of CSS classes exposed by this widget. To better understand the underlying DOM structure, have a look at the generated DOM in your browser.

Note that you can pass the prop `class-names`, with an object of class names and their replacement to override this.

Class name | Description
---|---
`ais-Panel` | The root div of the Panel
`ais-Panel--noRefinement` | The root div of the widget with no refinement
`ais-Panel-header` | The header of the Panel (optional)
`ais-Panel-body` | The body of the Panel
`ais-Panel-footer` | The footer of the Panel (optional)
