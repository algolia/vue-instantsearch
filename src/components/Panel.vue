<template>
  <div :class="[suit(), !canRefine && suit('', 'noRefinement')]">
    <div
      v-if="$slots.header || $scopedSlots.header"
      :class="suit('header')"
    >
      <slot
        name="header"
        :has-refinements="canRefine"
      />
    </div>
    <div :class="suit('body')">
      <slot :has-refinements="canRefine" />
    </div>
    <div
      v-if="$slots.footer || $scopedSlots.footer"
      :class="suit('footer')"
    >
      <slot
        name="footer"
        :has-refinements="canRefine"
      />
    </div>
  </div>
</template>

<script>
import { createPanelProviderMixin } from '../mixins/panel';
import { createSuitMixin } from '../mixins/suit';

export default {
  name: 'AisPanel',
  mixins: [createSuitMixin({ name: 'Panel' }), createPanelProviderMixin()],
};
</script>
