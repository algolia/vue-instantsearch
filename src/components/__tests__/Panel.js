import { mount } from '@vue/test-utils';
import Panel from '../Panel.vue';

describe('default render', () => {
  const defaultSlot = `
    <p>This is the body of the Panel.</p>
  `;

  it('renders correctly', () => {
    const wrapper = mount(Panel, {
      slots: {
        default: defaultSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly without refinement', () => {
    const wrapper = mount(Panel, {
      slots: {
        default: defaultSlot,
      },
    });

    wrapper.setData({
      canRefine: false,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with header', () => {
    const wrapper = mount(Panel, {
      slots: {
        default: defaultSlot,
        header: `<span>Header</span>`,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with footer', () => {
    const wrapper = mount(Panel, {
      slots: {
        default: defaultSlot,
        footer: `<span>Footer</span>`,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
