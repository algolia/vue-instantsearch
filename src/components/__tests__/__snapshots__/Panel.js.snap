// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render renders correctly 1`] = `

<div class="ais-Panel">
  <div class="ais-Panel-body">
    <p>
      This is the body of the Panel.
    </p>
  </div>
</div>

`;

exports[`default render renders correctly with footer 1`] = `

<div class="ais-Panel">
  <div class="ais-Panel-body">
    <p>
      This is the body of the Panel.
    </p>
  </div>
  <div class="ais-Panel-footer">
    <span>
      Footer
    </span>
  </div>
</div>

`;

exports[`default render renders correctly with header 1`] = `

<div class="ais-Panel">
  <div class="ais-Panel-header">
    <span>
      Header
    </span>
  </div>
  <div class="ais-Panel-body">
    <p>
      This is the body of the Panel.
    </p>
  </div>
</div>

`;

exports[`default render renders correctly without refinement 1`] = `

<div class="ais-Panel ais-Panel--noRefinement">
  <div class="ais-Panel-body">
    <p>
      This is the body of the Panel.
    </p>
  </div>
</div>

`;
